rocky/reports/views/report_overview.py

```diff
@@ -93,12 +96,119 @@ class ReportHistoryView(BreadcrumbsReportOverviewView, OctopoesView, ListView):
+            self.octopoes_api_connector.delete_many(all_report_ids, datetime.now(timezone.utc))
+        else:
+            self.octopoes_api_connector.delete_many(report_references, datetime.now(timezone.utc))
+        messages.success(self.request, _("Deletion successful."))
+
+    def rerun_reports(self, report_references: list[str]) -> None:
+        for report_id in report_references:
+            actual_report_ooi = self.get_report_ooi(report_id)
+
+            # First create new parent report and then create all subreports with new parent reference
+            if not actual_report_ooi.has_parent and actual_report_ooi.report_type == "concatenated-report":
+                new_concatenated_report = self.recreate_report(actual_report_ooi)
+                self.recreate_subreports(actual_report_ooi, new_concatenated_report)
+
+    def get_report_data_from_bytes(self, report: Report) -> dict[str, Any]:
+        self.bytes_client.login()
+        return TypeAdapter(Any, config={"arbitrary_types_allowed": True}).validate_json(
+            self.bytes_client.get_raw(raw_id=report.data_raw_id)
+        )
+
+    def recreate_report(self, report_ooi: Report, parent_report: Report | None = None) -> Report:
+        """
+        Creates a new report based on data of an old report.
+        If parent report is available then it is a subreport with a reference to the parent report.
+        """
+
+        observed_at = datetime.now(timezone.utc)
+
+        bytes_data = self.get_report_data_from_bytes(report_ooi)
+
+        new_bytes_id = self.bytes_client.upload_raw(
+            raw=ReportDataDict(bytes_data).model_dump_json().encode(), manual_mime_types={"openkat/report"}
+        )
+
+        new_report_ooi = Report(
+            name=report_ooi.name,
+            report_type=report_ooi.report_type,
+            template=report_ooi.template,
+            report_id=uuid4(),
+            organization_code=report_ooi.organization_code,
+            organization_name=report_ooi.organization_name,
+            organization_tags=report_ooi.organization_tags,
+            data_raw_id=new_bytes_id,
+            date_generated=observed_at,
+            input_oois=report_ooi.input_oois,
+            observed_at=observed_at,
+            parent_report=parent_report.reference if parent_report is not None else None,
+            has_parent=parent_report is not None,
+        )
+
+        create_ooi(self.octopoes_api_connector, self.bytes_client, new_report_ooi, observed_at)
+
+        return new_report_ooi
+
+    def recreate_subreports(self, report_ooi: Report, parent_ooi: Report) -> None:
+        subreports = self.octopoes_api_connector.query(
+            "Report.<parent_report[is Report]", valid_time=self.observed_at, source=report_ooi.reference
+        )
+        for subreport in subreports:
+            self.recreate_report(subreport, parent_ooi)
+
+    def rename_reports(self, report_references: list[str]) -> None:
```

Comment:
A few things:
- The messages here are a bit misleading, because there's no way to find out what message belongs to what report reference?
- On success you would spam a few "Report names changed successfully." messages on the view, you probably want to do that only once when it actually succeeds?


Also, the message "Report names changed successfully." is misleading